lib/ddtrace/contrib/http/patcher.rb

```diff
@@ -18,7 +18,7 @@ def should_skip_tracing?(req, address, port, transport, pin)
         # we don't want to trace our own call to the API (they use net/http)
         # when we know the host & port (from the URI) we use it, else (most-likely
         # called with a block) rely on the URL at the end.
-        if req.uri
+        if req.respond_to?(:uri) && req.uri
```

Comment:
Mmm, it's really a requirement at this point. We have plans to offer UDP connectivity for the agent (the main problem is the payload size, it's not uncommon to have traces which are 100k big, and need to be transferred at once) but for now installing our client lib requires `net/http` and uses it. Do you feel this is a problem?